tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -12420,14 +12420,27 @@ SpirvEmitter::processSpvIntrinsicCallExpr(const CallExpr *expr) {
 
   const auto args = expr->getArgs();
   for (uint32_t i = 0; i < expr->getNumArgs(); ++i) {
-    spvArgs.push_back(doExpr(args[i]));
+    auto param = funcDecl->getParamDecl(i);
+    const Expr *arg = args[i]->IgnoreParenLValueCasts();
+    SpirvInstruction *argInst = doExpr(arg);
+    if (param->hasAttr<VKReferenceExtAttr>()) {
```

Comment:
For example,
```
[[vk::ext_capability(5055)]]
[[vk::ext_extension("SPV_KHR_shader_clock")]]
[[vk::ext_instruction(/* OpReadClockKHR */ 5056)]]
uint64_t ReadClock([[vk::ext_reference]] uint scope);
...
clock = ReadClock(vk::DeviceScope);
```
we have to report the error for the above code because we cannot use `vk::DeviceScope` for a reference.

Reply:
thanks, that's very clear explanation.